src/jvm/main/org/jetbrains/kotlinx/lincheck/CTestStructure.java

```diff
@@ -154,6 +149,52 @@ private static void readTestStructureFromClass(Class<?> clazz, Map<String, Param
         }
     }
 
+    private static Map<String, ParameterGenerator<?>> createNamedGens(Class<?> clazz, RandomProvider randomProvider) {
+        Map<String, ParameterGenerator<?>> namedGens = new HashMap<>();
+        // Traverse all operations to determine EnumGens types or throw if one named enum gen is associated with many types
+        Map<String, Class<? extends Enum<?>>> enumGeneratorNameToClassMap = collectNamedEnumGeneratorToClassMap(clazz);
+        // Read named parameter paramgen (declared for class)
+        for (Param paramAnn : clazz.getAnnotationsByType(Param.class)) {
+            if (paramAnn.name().isEmpty()) {
```

Comment:
Use "when" instead of sever "if"s and document the code, please.

Reply:
To my mind, in this place we can't use java's switch properly to improve the code quality.